### ISSUE TITLE
Migrate to jwt-go v3

### ISSUE THREAD

**Issue by xiaowing** [influxdata/influxdb#6924]
With jwt-go being updated from v2 to v3 in the other day, building source code of v1.0.0beta2 will not end normally. The following usage of `token.Claims` will encounter build errors with the latest version of [jwt-go](https://github.com/dgrijalva/jwt-go)

I was wandering which version of jwt-go would be used at the release of  Influxdb v1.0.0 ?

With jwt-go v3, the build error will be like these:
```
# github.com/influxdata/influxdb/services/httpd
services/httpd/handler.go:793: invalid operation: token.Claims["exp"] (type jwt.Claims does not support indexing)
services/httpd/handler.go:799: invalid operation: token.Claims["username"] (type jwt.Claims does not support indexing)
```

The error occurred during building the following code

# services/httpd/handler.go
```go
if exp, ok := token.Claims["exp"].(float64); !ok || exp <= 0.0 { 
	h.httpError(w, "token expiration required", false, http.StatusUnauthorized) 
 	return 
} 
  
username, ok := token.Claims["username"].(string) 
if !ok { 
	h.httpError(w, "username in token must be a string", false, http.StatusUnauthorized) 
 	return 
} else if username == "" { 
 	h.httpError(w, "token must contain a username", false, http.StatusUnauthorized) 
 	return 
} 
```

According to the advice mentioned in the [MIGRATION GUIDE](https://github.com/dgrijalva/jwt-go/blob/master/MIGRATION_GUIDE.md). The source code should be changed like these:
```go
claims := token.Claims.(jwt.MapClaims)

if exp, ok := claims["exp"].(float64); !ok || exp <= 0.0 { 
	h.httpError(w, "token expiration required", false, http.StatusUnauthorized) 
 	return 
} 
  
username, ok := claims["username"].(string) 
if !ok { 
	h.httpError(w, "username in token must be a string", false, http.StatusUnauthorized) 
 	return 
} else if username == "" { 
 	h.httpError(w, "token must contain a username", false, http.StatusUnauthorized) 
 	return 
} 
```

Best regards.

**Comment by rossmcdonald** [comment]
You can see the pinned dependencies for InfluxDB [here](https://github.com/influxdata/influxdb/blob/master/Godeps), where jwt-go is currently pinned to [a2c85815a77d0f951e33ba4db5ae93629a1530af](https://github.com/dgrijalva/jwt-go/tree/a2c85815a77d0f951e33ba4db5ae93629a1530af). I don't expect we'll change it for v1.0 unless an issue is identified with the current implementation.

Please see the documentation [here](https://github.com/influxdata/influxdb/blob/master/CONTRIBUTING.md#build-and-test) regarding generating builds. It sounds like the build failure is due to `gdm restore` not being run.

**Comment by jsternberg** [comment]
I think @rossmcdonald is right and we shouldn't update the dependency until after 1.0. I'm going to tentatively put this in the 1.1 milestone and then we'll review if we are going to do it in that version when we start work on 1.1. Until now, we don't want to make any unnecessary changes to the currently released beta as we don't want to accidentally introduce a new bug.

**Comment by e-dard** [comment]
Fixed via #7136
